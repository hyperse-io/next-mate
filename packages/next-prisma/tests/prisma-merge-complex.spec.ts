import * as fs from 'node:fs';
import { promisify } from 'node:util';
import { prismaMerge } from '../src/index.js';

const readFile = promisify(fs.readFile);

describe('prisma-merge-complex', () => {
  test('merge prisma complex', async () => {
    const luciaPrismaContent = `
    model User {
      id  String  @id @default(uuid())
      name String
    }
`;

    await prismaMerge({
      luciaPrismaModel: luciaPrismaContent,
      input: ['tests/prisma/test3.prisma'],
      output: 'tests/prisma/testMerge3.prisma',
    });

    const prismaContent = await readFile('tests/prisma/testMerge3.prisma', {
      encoding: 'utf-8',
    });
    expect(prismaContent).toMatchSnapshot();
  });
});
