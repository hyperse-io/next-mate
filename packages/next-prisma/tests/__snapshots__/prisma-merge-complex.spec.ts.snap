// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prisma-merge-complex > merge prisma complex 1`] = `
"// *** GENERATED BY next-prisma :: DO NOT EDIT ***
model User {
	id String @id @default(uuid(4))
	username String @unique
	email String? @unique
	authSession Session[] @relation(name: "SessionToUser")
	authorized Authorized[] @relation(name: "AuthorizedToUser")
	createdAt DateTime @default(now())
	updatedAt DateTime @updatedAt
	name String 
	@@map("user")
}
model Authorized {
	id String @id @default(uuid(4))
	userId String 
	hashedPassword String? 
	providerId String? 
	providerMethod String 
	user User @relation(name: "AuthorizedToUser", fields: [userId], references: [id], onDelete: Cascade)
	createdAt DateTime @default(now())
	updatedAt DateTime @updatedAt
	@@index([userId])
	@@map("authorized")
}
model Session {
	id Int @id @default(autoincrement())
	token String @unique
	userId String 
	expiresAt DateTime @default(now())
	user User @relation(name: "SessionToUser", fields: [userId], references: [id], onDelete: Cascade)
	createdAt DateTime @default(now())
	updatedAt DateTime @updatedAt
	@@index([userId])
	@@map("session")
}"
`;
